crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -853,60 +851,106 @@ impl UncompressedDataManager {
+    use crate::DataFolders;
 
-    const CURRENT_BATCH_INDEX: u64 = 0;
     const UNIVARIATE_ID: u64 = 9674644176454356993;
 
     // Tests for UncompressedDataManager.
     #[tokio::test]
-    async fn test_can_find_existing_on_disk_data_buffers() {
-        // Spill an uncompressed buffer to disk.
+    async fn test_can_compress_existing_on_disk_data_buffers_when_initializing() {
         let temp_dir = tempfile::tempdir().unwrap();
-        let model_table_metadata = Arc::new(test::model_table_metadata());
-        let mut buffer = UncompressedInMemoryDataBuffer::new(
-            UNIVARIATE_ID,
-            model_table_metadata,
-            CURRENT_BATCH_INDEX,
+        let local_data_folder =
+            Arc::new(LocalFileSystem::new_with_prefix(temp_dir.path()).unwrap());
+
+        // Create a context with a storage engine.
+        let context = Arc::new(
+            Context::try_new(
+                Arc::new(Runtime::new().unwrap()),
+                &DataFolders {
+                    local_data_folder: local_data_folder.clone(),
+                    remote_data_folder: None,
+                    query_data_folder: local_data_folder,
+                },
+                ClusterMode::SingleNode,
+                ServerMode::Edge,
+            )
+            .await
+            .unwrap(),
         );
-        buffer.insert_data(CURRENT_BATCH_INDEX, 100, 10.0);
-        buffer
-            .spill_to_apache_parquet(temp_dir.path())
+
+        // Create a table in the context.
+        context
+            .parse_and_create_table(test::MODEL_TABLE_SQL, &context)
             .await
             .unwrap();
 
-        // The uncompressed data buffer should be referenced by the uncompressed data manager.
-        let (_metadata_manager, _data_manager, _model_table_metadata) =
-            create_managers(temp_dir.path()).await;
+        // Ingest a single data point and sleep to allow the ingestion thread to finish.
+        let mut storage_engine = context.storage_engine.write().await;
+        let model_table_metadata = Arc::new(test::model_table_metadata());
+        let data = uncompressed_data(1, model_table_metadata.schema.clone());
+
+        storage_engine
+            .insert_data_points(model_table_metadata, data)
+            .await
+            .unwrap();
+
+        sleep(Duration::from_millis(250)).await;
+
+        for _ in 0..2 {
+            storage_engine
+                .uncompressed_data_manager
+                .spill_in_memory_data_buffer()
+                .await
+                .unwrap();
+        }
 
-        // TODO: refactor tests to use a shared context with enough information for initialize().
-        //data_manager.initialize(context);
+        // Compress the spilled buffers and sleep to allow the compression thread to finish.
+        let result = storage_engine.initialize(&context).await;
+        assert!(result.is_ok());
+        sleep(Duration::from_millis(250)).await;
```

Comment:
Maybe check the result has been computed with a shorter sleep duration to not wait more than necessary and not make the test fail due to not sleeping long enough.

Reply:
After some manual testing 250 ms seems to be a good trade off between speed and making sure the slower Github build does not fail.